File: 17-pascal-triangle/pascal_tri.php
<?php
// PHP program to print Pascal Triangle
// Created by M.V.Harish Kumar on 18-09-2024

function fact($n) {
	if ($n <= 1) return 1;
	$prod = 2;
	for($i = 3; $i <= $n; $i++) $prod *= $i;
	return $prod;
}

function n_C_r($n, $r) {
	return intdiv(fact($n), fact($r)*fact($n-$r));
}

$n = intval(readline("Enter no. of rows: "));
for ($i = 0; $i < $n; $i++) {
	for($_ = 0; $_ <= ($n-$i); $_++) echo " ";
	for($j = 0; $j <= $i; $j++) echo n_C_r($i, $j) . " ";
	echo "\n";
}

File: 17-pascal-triangle/pascal_tri.go
// Go program to print pascal triangle
// Written by M.V.Harish Kumar on 18-06-2025
package main

import (
	"fmt"
	"strings"
)

func fact(n uint) uint {
	var res uint = 1
	for i := range n {
		res *= i + 1
	}
	return res
}

func nCr(n, r uint) uint {
	return fact(n) / (fact(r) * fact(n-r))
}

func main() {
	var nterms uint
	fmt.Print("Enter no of rows to print: ")
	fmt.Scanln(&nterms)

	for i := range nterms {
		fmt.Print(strings.Repeat(" ", int(nterms - i - 1)))
		for j := range i + 1 {
			fmt.Print(nCr(i, j), " ")
		}
		fmt.Println()
	}
}
